src/Analyzers/CSharp/Tests/UseImplicitObjectCreation/CSharpUseImplicitObjectCreationTests.cs

```diff
@@ -161,5 +177,74 @@ class C
                 LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
             }.RunAsync();
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitObjectCreation)]
+        public async Task TestNotWithTypeParameter()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+class C<T> where T : new()
+{
+    T t = new [|T|]();
+}",
+                FixedCode = @"
+class C<T> where T : new()
+{
+    T t = new();
+}",
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitObjectCreation)]
+        public async Task TestWithLocalWhenUserDoesNotPreferVar()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+class C
+{
+    void M()
+    {
+        C c = new [|C|]();
+    }
+}",
+                FixedCode = @"
+class C
+{
+    void M()
+    {
+        C c = new();
+    }
+}",
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                Options =
+                {
+                    { CSharpCodeStyleOptions.VarWhenTypeIsApparent, CodeStyleOptions2.FalseWithSuggestionEnforcement },
```

Comment:
💡 I believe you can also just set the "value" for it:

```suggestion
                    { CSharpCodeStyleOptions.VarWhenTypeIsApparent, false },
```

There is also a three-argument overload

```suggestion
                    { CSharpCodeStyleOptions.VarWhenTypeIsApparent, false, NotificationOption2.Suggestion },
```